core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -4852,6 +4853,42 @@ class DAGSchedulerSuite extends SparkFunSuite with TempLocalSparkContext with Ti
     assert(mapStatuses.count(s => s != null && s.location.executorId == "hostB-exec") === 1)
   }
 
+  /**
+   * In this test, we simulate a job where some tasks in a stage fail, and it triggers the retry
+   * of the task in its previous stage. The two attempts of the same task in the previous stage
+   * produce different shuffle checksums.
+   */
+  test("Output usage log for tasks that produce different checksum across retries") {
```

Comment:
what does this test? I thought we will check `MapOutputTracker.checksumMismatchIndices`

Reply:
you are right. it's supposed to check the `checksumMismatchIndices`. updated.